File: Problem-049/solution.py
import math


def prime_permutations():

    def is_prime(n):
        if n < 2:
            return False

        for i in range(2, math.isqrt(n) + 1):
            if n % i == 0:
                return False

        return True


    def are_permutations(a, b, c):
        return (
            sorted(str(a)) ==
            sorted(str(b)) ==
            sorted(str(c))
        )


    for a in range(1000, 10000):

        if not is_prime(a):
            continue

        for b in range(a + 1, 10000):

            if not is_prime(b):
                continue

            # Common difference
            difference = b - a

            # Third number in arithmetic sequence
            c = b + difference

            if c >= 10000:
                break

            if is_prime(c) and are_permutations(a, b, c):

                # Skip the example sequence
                # 1487, 4817, 8147
                if a == 1487:
                    continue

                return int(str(a) + str(b) + str(c))


print(prime_permutations())

File: Problem-049/solution.js
function primePermutations() {

    function isPrime(n) {
        if (n < 2) return false;

        for (let i = 2; i * i <= n; i++) {
            if (n % i === 0) {
                return false;
            }
        }

        return true;
    }


    function arePermutations(a, b, c) {
        const x = a.toString().split('').sort().join('');
        const y = b.toString().split('').sort().join('');
        const z = c.toString().split('').sort().join('');

        return x === y && y === z;
    }


    for (let a = 1000; a < 10000; a++) {

        if (!isPrime(a)) continue;

        for (let b = a + 1; b < 10000; b++) {

            if (!isPrime(b)) continue;

            let difference = b - a;
            let c = b + difference;

            if (c >= 10000) break;

            if (
                isPrime(c) &&
                arePermutations(a, b, c)
            ) {

                // Skip the example given in the question
                if (a === 1487) continue;

                return Number(
                    a.toString() +
                    b.toString() +
                    c.toString()
                );
            }
        }
    }
}


console.log(primePermutations());
